### ISSUE TITLE
nodes with different monitor interfaces complain about invalid dict 

### ISSUE THREAD

**Issue by vasukulkarni** [ceph/ceph-ansible#1660]
inventory is like below, nodes have different interfaces for monitor.

```
2017-07-06T16:41:38.051 INFO:teuthology.orchestra.run.smithi019:Running: 'cat ceph-ansible/inven.yml'
2017-07-06T16:41:38.107 INFO:teuthology.orchestra.run.smithi019.stdout:[clients]
2017-07-06T16:41:38.108 INFO:teuthology.orchestra.run.smithi019.stdout:smithi031.front.sepia.ceph.com devices='["/dev/nvme0n1p1", "/dev/nvme0n1p2", "/dev/nvme0n1p3"]' monitor_interface='ens1f0' public_network='172.21.0.0/20'
2017-07-06T16:41:38.108 INFO:teuthology.orchestra.run.smithi019.stdout:
2017-07-06T16:41:38.108 INFO:teuthology.orchestra.run.smithi019.stdout:[mdss]
2017-07-06T16:41:38.108 INFO:teuthology.orchestra.run.smithi019.stdout:smithi019.front.sepia.ceph.com devices='["/dev/nvme0n1p1", "/dev/nvme0n1p2", "/dev/nvme0n1p3"]' monitor_interface='ens1f0' public_network='172.21.0.0/20'
2017-07-06T16:41:38.108 INFO:teuthology.orchestra.run.smithi019.stdout:
2017-07-06T16:41:38.108 INFO:teuthology.orchestra.run.smithi019.stdout:[mgrs]
2017-07-06T16:41:38.108 INFO:teuthology.orchestra.run.smithi019.stdout:smithi031.front.sepia.ceph.com devices='["/dev/nvme0n1p1", "/dev/nvme0n1p2", "/dev/nvme0n1p3"]' monitor_interface='ens1f0' public_network='172.21.0.0/20'
2017-07-06T16:41:38.108 INFO:teuthology.orchestra.run.smithi019.stdout:smithi144.front.sepia.ceph.com devices='["/dev/nvme0n1p1", "/dev/nvme0n1p2", "/dev/nvme0n1p3"]' monitor_interface='enp3s0f1' public_network='172.21.0.0/20'
2017-07-06T16:41:38.108 INFO:teuthology.orchestra.run.smithi019.stdout:
2017-07-06T16:41:38.109 INFO:teuthology.orchestra.run.smithi019.stdout:[mons]
2017-07-06T16:41:38.109 INFO:teuthology.orchestra.run.smithi019.stdout:smithi019.front.sepia.ceph.com devices='["/dev/nvme0n1p1", "/dev/nvme0n1p2", "/dev/nvme0n1p3"]' monitor_interface='ens1f0' public_network='172.21.0.0/20'
2017-07-06T16:41:38.109 INFO:teuthology.orchestra.run.smithi019.stdout:smithi031.front.sepia.ceph.com devices='["/dev/nvme0n1p1", "/dev/nvme0n1p2", "/dev/nvme0n1p3"]' monitor_interface='ens1f0' public_network='172.21.0.0/20'
2017-07-06T16:41:38.109 INFO:teuthology.orchestra.run.smithi019.stdout:smithi144.front.sepia.ceph.com devices='["/dev/nvme0n1p1", "/dev/nvme0n1p2", "/dev/nvme0n1p3"]' monitor_interface='enp3s0f1' public_network='172.21.0.0/20'
2017-07-06T16:41:38.109 INFO:teuthology.orchestra.run.smithi019.stdout:
2017-07-06T16:41:38.109 INFO:teuthology.orchestra.run.smithi019.stdout:[osds]
2017-07-06T16:41:38.109 INFO:teuthology.orchestra.run.smithi019.stdout:smithi019.front.sepia.ceph.com devices='["/dev/nvme0n1p1", "/dev/nvme0n1p2", "/dev/nvme0n1p3"]' monitor_interface='ens1f0' public_network='172.21.0.0/20'
2017-07-06T16:41:38.109 INFO:teuthology.orchestra.run.smithi019.stdout:smithi031.front.sepia.ceph.com devices='["/dev/nvme0n1p1", "/dev/nvme0n1p2", "/dev/nvme0n1p3"]' monitor_interface='ens1f0' public_network='172.21.0.0/20'
2017-07-06T16:41:38.109 INFO:teuthology.orchestra.run.smithi019.stdout:smithi144.front.sepia.ceph.com devices='["/dev/nvme0n1p1", "/dev/nvme0n1p2", "/dev/nvme0n1p3"]' monitor_interface='enp3s0f1' public_network='172.21.0.0/20'
2017-07-06T16:41:38.110 INFO:teuthology.orchestra.run.smithi019:Running: 'cat ceph-ansible/site.yml'
```

we get the below failure

```
2017-07-06T16:46:10.975 INFO:teuthology.orchestra.run.smithi019.stdout:MSG:
2017-07-06T16:46:10.975 INFO:teuthology.orchestra.run.smithi019.stdout:
2017-07-06T16:46:10.975 INFO:teuthology.orchestra.run.smithi019.stdout:'dict object' has no attribute u'ansible_ens1f0'
2017-07-06T16:46:10.977 INFO:teuthology.orchestra.run.smithi019.stdout:fatal: [smithi144.front.sepia.ceph.com]: FAILED! => {
2017-07-06T16:46:10.977 INFO:teuthology.orchestra.run.smithi019.stdout:    "failed": true
2017-07-06T16:46:10.978 INFO:teuthology.orchestra.run.smithi019.stdout:}
2017-07-06T16:46:10.978 INFO:teuthology.orchestra.run.smithi019.stdout:
2017-07-06T16:46:10.978 INFO:teuthology.orchestra.run.smithi019.stdout:MSG:
2017-07-06T16:46:10.978 INFO:teuthology.orchestra.run.smithi019.stdout:
2017-07-06T16:46:10.978 INFO:teuthology.orchestra.run.smithi019.stdout:'dict object' has no attribute u'ansible_enp3s0f1'
2017-07-06T16:46:10.987 INFO:teuthology.orchestra.run.smithi019.stdout:fatal: [smithi031.front.sepia.ceph.com]: FAILED! => {
2017-07-06T16:46:10.987 INFO:teuthology.orchestra.run.smithi019.stdout:    "failed": true
2017-07-06T16:46:10.987 INFO:teuthology.orchestra.run.smithi019.stdout:}
2017-07-06T16:46:10.987 INFO:teuthology.orchestra.run.smithi019.stdout:
2017-07-06T16:46:10.987 INFO:teuthology.orchestra.run.smithi019.stdout:MSG:
2017-07-06T16:46:10.987 INFO:teuthology.orchestra.run.smithi019.stdout:
2017-07-06T16:46:10.988 INFO:teuthology.orchestra.run.smithi019.stdout:'dict object' has no attribute u'ansible_ens1f0'
2017-07-06T16:46:10.988 INFO:teuthology.orchestra.run.smithi019.stdout:
2017-07-06T16:46:10.988 INFO:teuthology.orchestra.run.smithi019.stdout:PLAY RECAP *********************************************************************
2017-07-06T16:46:10.988 INFO:teuthology.orchestra.run.smithi019.stdout:smithi019.front.sepia.ceph.com : ok=43   changed=12   unreachable=0    failed=1
2017-07-06T16:46:10.989 INFO:teuthology.orchestra.run.smithi019.stdout:smithi031.front.sepia.ceph.com : ok=43   changed=11   unreachable=0    failed=1
2017-07-06T16:46:10.989 INFO:teuthology.orchestra.run.smithi019.stdout:smithi144.front.sepia.ceph.com : ok=41   changed=11   unreachable=0    failed=1
```

**Comment by vasukulkarni** [comment]
assigning this to @andrewschoen since his private branch run fixed this issue.

**Comment by vasukulkarni** [comment]
http://pulpito.ceph.com/vasu-2017-07-06_18:54:09-ceph-ansible-master-distro-basic-smithi/1367701/, the test failed due to fsid issue which is different

**Comment by guits** [comment]
fixed by #1666
